drivers/include/disp_dev.h

```diff
@@ -42,23 +44,87 @@ extern "C" {
  */
 typedef struct disp_dev disp_dev_t;
 
+/**
+ * @brief Type to store the update coordinates in
+ */
+typedef struct {
+    uint16_t x1; /**< Left coordinate */
+    uint16_t x2; /**< Right coordinate */
+    uint16_t y1; /**< Top coordinate */
+    uint16_t y2; /**< Bottom coordinate */
+} disp_dev_coordinates_t;
+
+/**
+ * @brief Type for color depth.
+ */
+typedef enum {
+    MONOCHROME = 0,
+    2_BIT_COLOR,
+    GRAYSCALE,
+    8_BIT_COLOR,
+    16_BIT_HIGH_COLOR,
+    24_BIT_TRUE_COLOR
+} disp_dev_color_depth_t;
+
+/**
+ * @brief Type to hold a pixel for monochrome data
+ */
+typedef struct {
+    bool color;
+} disp_dev_pix_mono_t;
+
+/**
+ * @brief Type to hold a pixel for 2bit color data
+ */
+typedef struct {
```

Comment:
Not sure about this though. Thought about storing it in an `uint8_t` instead of two booleans. But maybe this way the compiler is able to optimize it better.